odoo/addons/base/models/res_currency.py

```diff
@@ -97,6 +97,15 @@ def _deactivate_group_multi_currency(self):
         if group_user and group_mc:
             group_user.sudo()._remove_group(group_mc.sudo())
 
+    @api.constrains('active')
+    def _check_company_currency_stays_active(self):
+        if self._context.get('install_mode') or self._context.get('force_deactivate'):
+            return
+
+        currencies = self.filtered(lambda c: not c.active)
+        if self.env['res.company'].search([('currency_id', 'in', currencies.ids)]):
+            raise UserError(_("This currency is set on a company and therefore must be active."))
```

Comment:
just a side note: not sure 100% about the English, actually. 

maybe: 
This currency is set on a company and therefore must be kept active.
or:
This currency is set on a company and therefore cannot be deactivated.

would have been better? What do you think? :thinking:

Reply:
I t did not shock me before, but sure, why not. Changed.

Reply:
Indeed. Done.